lib/ansible/modules/cloud/openstack/os_tempest_run.py

```diff
@@ -0,0 +1,296 @@
+        required: False
+        default: None
+    force:
+        description:
+            When C(True) override output file if already exists
+        required: False
+        default: 'False'
+    subunit:
+        description:
+            When C(True) show the output as subunit v2
+        required: False
+        default: 'False'
+notes:
+    - You can find out more about Tempest at U(http://docs.openstack.org/developer/tempest/)
+    - The module requires to tempest init <I(workspace)> before usage
+    - The options I(whitelist_file) and I(blacklist_file) are mutually exclusive, if they are both given only the I(whitelist_file) will be used
+'''
+
+EXAMPLES = '''
+# Run all tests with default number of workers
+- os_tempest_run:
+    dest: DEST
+    workspace: cloud
+
+# Run all tests matching to a regex with default number of workers
+- os_tempest_run:
+    dest: DEST
+    workspace: cloud
+    regex: REGEX
+
+# Run all tests serially
+- os_tempest_run:
+    dest: DEST
+    workspace: cloud
+    concurrency: 1
+
+# Run all tests with 4 workers
+- os_tempest_run:
+    dest: DEST
+    workspace: cloud
+    concurrency: 4
+
+# Run all tests that their REGEX is in the whitelist file
+- os_tempest_run:
+    dest: DEST
+    workspace: cloud
+    whitelist_file: /path/to/whitelist
+
+# Run all tests that their REGEX is not in the blacklist file
+- os_tempest_run:
+    dest: DEST
+    workspace: cloud
+    blacklist_file: /path/to/blacklist
+'''
+RETURN = '''
+stdout:
+    description: Tempest's stdout
+    returned: fail
+    type: string
+    sample: ""
+stderr:
+    description: Tempest's stderr
+    returned: fail
+    type: string
+    sample: ""
+command:
+    description: the command executed to run Tempest
+    returned: fail
+    type: string
+    sample: "tempest run --workspace cloud"
+output:
+    description: path to the file containing the output from Tempest
+    returned: success
+    type: string
+    sample: "/path/to/file.subunit"
+'''
+from ansible.module_utils.basic import AnsibleModule
+import os
+import sys
+
+# imports for the code copied from ansible.utils.path
+from errno import EEXIST
+from ansible.module_utils._text import to_bytes
+from ansible.module_utils._text import to_native
+from ansible.module_utils._text import to_text
+
+
+def main():
+    ansible_module = AnsibleModule(argument_spec=dict(
+        workspace=dict(type="str", required=True),
+        dest=dict(type="path", required=True),
+        regex=dict(type="str", required=False, default=""),
+        whitelist_file=dict(type="path", required=False, default=""),
+        blacklist_file=dict(type="path", required=False, default=""),
+        concurrency=dict(type="int", required=False, default=None),
+        force=dict(type="bool", required=False, default=False),
+        subunit=dict(type="bool", required=False, default=True),
+    ))
```

Comment:
There are some options to `AnsibleModule` which you may be able to use to validate options passed in.
Look at existing modules for examples:
```
mutually_exclusive
required_together
required_one_of
require_if
```


Should `whitelist_file` and `blacklist_file` be `mutually_exclusive`
`mutually_exclusive=(('blacklist_file', 'whitelist_file'),),`